src/pip/_internal/operations/build/metadata_legacy.py

```diff
@@ -5,77 +5,38 @@
 import os
 
 from pip._internal.exceptions import InstallationError
-from pip._internal.utils.misc import ensure_dir
 from pip._internal.utils.setuptools_build import make_setuptools_egg_info_args
 from pip._internal.utils.subprocess import call_subprocess
+from pip._internal.utils.temp_dir import TempDirectory
 from pip._internal.utils.typing import MYPY_CHECK_RUNNING
-from pip._internal.vcs import vcs
 
 if MYPY_CHECK_RUNNING:
-    from typing import List, Optional
-
     from pip._internal.build_env import BuildEnvironment
 
 logger = logging.getLogger(__name__)
 
 
-def _find_egg_info(source_directory, is_editable):
-    # type: (str, bool) -> str
-    """Find an .egg-info in `source_directory`, based on `is_editable`.
+def _find_egg_info(source_directory):
```

Comment:
```suggestion
def _find_egg_info(directory):
```

Reply:
The function body and docstring also need to match this change :)